### PR TITLE
Add Luhn algorithm

### PR DESCRIPTION
**Pull request by NikitaYermak**
Add Luhn algorithm. It`s a simple checksum formula used to validate a variety of identification numbers, such as credit card numbers. More information on the link: https://en.wikipedia.org/wiki/Luhn_algorithm.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the comments
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Other/Luhn.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        /// <returns>True: Number is valid.
+        /// False: Number isn`t valid.</returns>
+        public static bool Validate(string number)
+        {
+            if (GetSum(number) % 10 == 0)
+            {
+                return true;
+            }
+
+            return false;
```
Comment (siriak) [inline Algorithms/Other/Luhn.cs:28]:
```suggestion
            return GetSum(number) % 10 == 0;
```

File: Algorithms/Other/Luhn.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        /// <summary>
+        /// This algorithm only finds one number.
+        /// In place of the unknown digit, put "x".
+        /// </summary>
+        /// <param name="number">The number in which to find the missing digit.</param>
+        /// <returns>Missing digit.</returns>
+        public static int GetLostNum(string number)
+        {
+            int lostIndex = Reverse(number).IndexOf("x", StringComparison.CurrentCultureIgnoreCase);
```
Comment (siriak) [inline Algorithms/Other/Luhn.cs:39]:
```suggestion
            int lostIndex = number.Length - 1 - number.LastIndexOf("x", StringComparison.CurrentCultureIgnoreCase);
```

File: Algorithms/Other/Luhn.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        }
+
+        /// <summary>
+        /// The sum found by the algorithm.
+        /// </summary>
+        /// <param name="number">The number for which the amount will be found.</param>
+        /// <returns>Sum.</returns>
+        private static int GetSum(string number)
+        {
+            string reverseNumber = Reverse(number);
```
Comment (siriak) [inline Algorithms/Other/Luhn.cs:64]: Can you just walk from the end in the loop below?

File: Algorithms/Other/Luhn.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            {
+                return lostNum;
+            }
+            else
+            {
+                int tempLostNum = lostNum / 2;
+
+                // Case 2: if the index of the lost digit isn`t even and that number <= 4.
+                // Case 3: if the index of the lost digit isn`t even and that number > 4.
+                return Validate(number.Replace("x", tempLostNum.ToString())) ? tempLostNum : (lostNum + 9) / 2;
```
Comment (siriak) [inline Algorithms/Other/Luhn.cs:53]: Could you decide on which number to add without validation?

File: Algorithms/Other/Luhn.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                temp = reverseNumber[i] - '0';
+                if (i % 2 != 0)
+                {
+                    temp *= 2;
+                    if (temp > 9)
+                    {
+                        temp -= 9;
+                    }
+
+                    sum += temp;
```
Comment (siriak) [inline Algorithms/Other/Luhn.cs:78]:
```suggestion
                    sum += (2 * temp) % 9;
```

**Review by siriak — CHANGES_REQUESTED** [review]
Looks good, I've left some comments though

**Review by siriak — APPROVED** [review]
Thanks!

**Comment by NikitaYermak** [comment]
Thanks for improving my code.  I will add some more algorithm soon.

**Pull request merged**